### PR TITLE
OpenNem historical

### PR DESCRIPTION
**Pull request by corradio**
Adds ability to refetch historical data
Tested locally with production + interconnectors + price (realtime and historical)

### CODE REVIEW

File: parsers/OPENNEM.py [hunk 1]
```diff
@@ -75,48 +85,70 @@ def dataset_to_df(dataset):
...
-            for obj in filtered_datasets if obj['region'] == ZONE_KEY_TO_REGION[zone_key]
+            for obj in filtered_datasets if obj['region'] == region
         ])
         return df, pd.Series(capacities)
     else:
         return df
 
 
-def sum_vector(pd_series, keys, transform=lambda x: x):
+def sum_vector(pd_series, keys, transform=None):
```
Comment (corradio) [inline parsers/OPENNEM.py:142]: Made this function a bit faster using pands

File: parsers/OPENNEM.py [hunk 2]
```diff
@@ -163,9 +196,10 @@ def fetch_production(zone_key=None, session=None, target_datetime=None, logger=l
         },
         'source': SOURCE,
         'zoneKey': zone_key,
-    } for dt, row in df.iterrows()]
+    } for dt, row in tqdm(df.iterrows())]
```
Comment (corradio) [inline parsers/OPENNEM.py:199]: Used tqdm to print the speed of this iteration (it's currently the bottleneck in the parser)

File: test_parser.py [hunk 3]
```diff
@@ -21,6 +21,7 @@
 )
 
 logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.DEBUG)
```
Comment (corradio) [inline test_parser.py:24]: set default logging to DEBUG in test_parser so we can see all debug logs

File: pyproject.toml [hunk 4]
```diff
@@ -29,6 +29,7 @@ opencv-python = {version="4.2.0.32", optional=true}
 xlrd = {version="1.2.0", optional=true}
 freezegun = {version="0.3.15", optional=true}
 signalr-client-threads = {version="0.0.12", optional=true}
+tqdm = "^4.59.0"
```
Comment (skovhus) [inline pyproject.toml:32]: As this is only used by the parsers, we should add the `optional` flag here. I also lowered the version here to support the version we are using internally.

```suggestion
tqdm = {version="^4.19.5", optional=true}
```

We should also add `tqdm` to the list to `parsers` in the `[tool.poetry.extras]` section.

**Review by skovhus — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 4 above)
Comment (skovhus) [reply pyproject.toml]: > We should also add tqdm to the parsers list in the [tool.poetry.extras] section.

Reminder to update the pyproject.toml file.

Comment (skovhus) [reply pyproject.toml]: @corradio

Comment (corradio) [reply pyproject.toml]: Let me know if https://github.com/tmrowco/electricitymap-contrib/pull/3066/commits/407f0117520b9381178042311a4942b733299efe fixes it

**Review by skovhus — APPROVED** [review]

**Pull request merged**